Library/DasharoSystemFeaturesUiLib/DasharoSystemFeatures.c

```diff
@@ -487,14 +477,20 @@ DasharoSystemFeaturesRouteConfig (
       );
   ASSERT_EFI_ERROR (Status);
 
-  if (PrivateData->HybridCpuArchitecture) {
+  if (PrivateData->HybridCpuArchitecture && PcdGetBool (PcdShowCpuCoreDisable)) {
     if (DasharoFeaturesData.SmallCoreActiveCount == 0 && PrivateData->BigCoreMaxCount == 0)
       return EFI_INVALID_PARAMETER;
 
     if (DasharoFeaturesData.BigCoreActiveCount == 0 && PrivateData->SmallCoreMaxCount == 0)
       return EFI_INVALID_PARAMETER;
   }
 
+  // Can we just return EFI_SUCCESS? Or maybe EFI_UNSUPPORTED is more appropriate?
+  if (!PcdGetBool (PcdShowMenu)) {
+    Private->DasharoFeaturesData = DasharoFeaturesData;
+    return EFI_SUCCESS;
+  }

```

Comment:
If `PcdShowMenu` is false, this function can't be called because HiiConfigAccess protocol is never instantiated (see constructor).

Reply:
Removed